### ISSUE TITLE
Float literal pattern is an unexpected pattern

### ISSUE THREAD

**Issue by philderbeast** [unisonweb/unison#2194]
Please add tests for [float literal patterns](https://www.unisonweb.org/docs/language-reference/#literal-patterns).

https://github.com/unisonweb/unison/blob/e8dd1d5b7d94857bf5ba7a5e22736bc3010c3162/unison-src/tests/methodical/pattern-matching.u#L26-L33

```
matchZero : Float -> Boolean
matchZero = cases
    0.0 -> true
    _ -> false

anfInitCase: unexpected pattern: Float 0.0
```

```
matchZero : Float -> Boolean
matchZero x =
    match x with
    0.0 -> true
    _ -> false

anfInitCase: unexpected pattern: Float 0.0
```

**Comment by philderbeast** [comment]
A workaround for this is a guard pattern.

**Comment by runarorama** [comment]
Float patterns should probably be disallowed with a good error message, as checking Floats for equality is very likely to lead to bugs.

**Comment by dolio** [comment]
Addressed by #2299
